apps/server/src/modules/oauth/controller/oauth-sso.controller.ts

```diff
@@ -1,150 +1,18 @@
-import { Configuration } from '@hpi-schul-cloud/commons/lib';
-import {
-	Controller,
-	Get,
-	InternalServerErrorException,
-	Param,
-	Query,
-	Req,
-	Res,
-	Session,
-	UnauthorizedException,
-	UnprocessableEntityException,
-} from '@nestjs/common';
-import { ApiOkResponse, ApiResponse, ApiTags } from '@nestjs/swagger';
-import { ISession } from '@shared/domain/types/session';
+import { Authenticate, CurrentUser, ICurrentUser } from '@modules/authentication';
+import { Controller, Get, Param, Query, Req, UnauthorizedException } from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
 import { LegacyLogger } from '@src/core/logger';
-import { ICurrentUser, Authenticate, CurrentUser, JWT } from '@modules/authentication';
-import { OAuthMigrationError } from '@modules/user-login-migration/error/oauth-migration.error';
-import { MigrationDto } from '@modules/user-login-migration/service/dto';
-import { CookieOptions, Request, Response } from 'express';
-import { HydraOauthUc } from '../uc/hydra-oauth.uc';
-import { UserMigrationResponse } from './dto/user-migration.response';
-import { OAuthSSOError } from '../loggable/oauth-sso.error';
+import { Request } from 'express';
 import { OAuthTokenDto } from '../interface';
-import { OauthLoginStateMapper } from '../mapper/oauth-login-state.mapper';
-import { UserMigrationMapper } from '../mapper/user-migration.mapper';
-import { OAuthProcessDto } from '../service/dto';
-import { OauthUc } from '../uc';
-import { OauthLoginStateDto } from '../uc/dto/oauth-login-state.dto';
-import { AuthorizationParams, SSOLoginQuery, SystemIdParams } from './dto';
+import { HydraOauthUc } from '../uc';
+import { AuthorizationParams } from './dto';
 import { StatelessAuthorizationParams } from './dto/stateless-authorization.params';
 
 @ApiTags('SSO')
 @Controller('sso')
 export class OauthSSOController {
-	private readonly clientUrl: string;
-
-	constructor(
-		private readonly oauthUc: OauthUc,
```

Comment:
OauthUc is no longer needed and modules have to be changed. Also the oauth service contains unused code now.

Reply:
also delete all unused mappers and interfaces like the request params.

Reply:
Otherwise, I have removed most of the things in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4548 too